src/core/ddsi/src/ddsi_plist.c

```diff
@@ -3615,6 +3615,42 @@ dds_qos_t * ddsi_xqos_dup (const dds_qos_t *src)
   return dst;
 }
 
+bool ddsi_xqos_add_property_if_unset (dds_qos_t *q, bool propagate, const char *name, const char *value)
```

Comment:
This is really similar to `dds_qset_prop`, the difference being that this one doesn't replace the value. Perhaps it would make more sense to generalize `dds_qset_prop` instead?

Reply:
I don't know about this one. All the dds_qset/get ops are nicely contained in one place, if I generalise something I don't know what the logical place is to put the definition and implementation in a way that makes the function accessible from both dds and ddsi. Are you okay with leaving this semi-duplicate implementation or do you have a suggestion where to put the definitions for the generalised one?